### PR TITLE
Set symbols as default

### PR DESCRIPTION
**Pull request by vlt1**
Added new Annotations plugin props to handle default point type and some symbol style properties.

## Description

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [ ] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [x] Other... Please describe: enhancement

## Issue
#5452
[austrocontrol-C125/#167](https://github.com/geosolutions-it/austrocontrol-C125/issues/167)

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

**Comment by MV88** [comment]
please:
- open an issue in this report that refers to the other one 
- connect this pr to the newly created issue.
- update the current behaviour with the link to the newly created issue

**Comment by MV88** [comment]
hi @vlt1 can you please resolve the conflicts?

File: web/client/components/style/vector/Manager.jsx [hunk 1]
```diff
@@ -211,68 +206,7 @@ class Manager extends React.Component {
         this.props.onChangeStyle(newStyles);
     }
     changeSymbolType = (id, pointType) => {
-        // TODO FIX THIS
-
-        let defaultSymbolStyle = {};
```
Comment (MV88) [inline web/client/components/style/vector/Manager.jsx:216]: are you sure this is no longer needed?
also why you moved to annotations epic?

File: web/client/epics/annotations.js [hunk 2]
```diff
@@ -699,6 +703,44 @@ module.exports = (viewer) => ({
                 addClickCallback: true
             }, assign({}, style, {highlight: false}));
             return Rx.Observable.of(action);
+        }),
+    loadDefaultAnnotationsStylesEpic: (action$, store) => action$.ofType(LOAD_DEFAULT_STYLES)
```
Comment (MV88) [inline web/client/epics/annotations.js:707]: can you add a little of documentation?

File: web/client/components/mapcontrols/annotations/Annotations.jsx [hunk 3]
```diff
@@ -83,6 +84,10 @@ const defaultConfig = require('./AnnotationsConfig');
  * @prop {string} symbolsPath path to the svg folder
  * @prop {object[]} symbolList list of symbols
  * @prop {string} defaultShape default Shape
+ * @prop {string} defaultShapeStrokeColor default symbol stroke color
+ * @prop {string} defaultShapeFillColor default symbol fill color
+ * @prop {string} defaultShapeSize default symbol shape size in px
+ * @prop {object} defaultStyles object with default symbol styles
```
Comment (MV88) [inline web/client/components/mapcontrols/annotations/Annotations.jsx:90]: here it is missing defaultPointType, please add also values that can be used symbol or marker

File: web/client/components/mapcontrols/annotations/AnnotationsEditor.jsx [hunk 4]
```diff
@@ -102,8 +102,13 @@ const bbox = require('@turf/bbox');
  * @prop {string} pointType the type of the point, values are "marker" or "symbol"
  * @prop {object[]} lineDashOptions list of options for dashed lines
  * @prop {object[]} symbolList list of symbols
- * @prop {string} defaultShape default shape for symbol
  * @prop {string} symbolsPath path to the svg folder
+ * @prop {string} defaultShape default shape for symbol
+ * @prop {string} defaultShapeStrokeColor default symbol stroke color
+ * @prop {string} defaultShapeFillColor default symbol fill color
+ * @prop {string} defaultShapeSize default symbol shape size in px
+ * @prop {string} defaultPointType default point type
```
Comment (MV88) [inline web/client/components/mapcontrols/annotations/AnnotationsEditor.jsx:110]: please add values "symbol" or "marker"
```suggestion
 * @prop {string} defaultPointType default point type, can be "symbol" or "marker", default is "marker"
```

File: web/client/plugins/Annotations.jsx [hunk 5]
```diff
@@ -194,7 +195,10 @@ const conditionalToggle = on.bind(null, toggleControl('annotations', null), (sta
   * If an odd number of values is inserted then they are added again to reach an even number of values
   * for more information see [this page](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/stroke-dasharray)
   * @prop {string} symbolsPath the relative path to the symbols folder where symbols.json and SVGs are located (starting from the index.html folder, i.e. the root)
-  * @prop {string} defaultShape the default symbol used when switching to the symbol styler from marker styler
+  * @prop {string} defaultShape the default symbol used when switching for the symbol styler
+  * @prop {string} defaultShapeStrokeColor default symbol stroke color
+  * @prop {string} defaultShapeFillColor default symbol fill color
+  * @prop {string} defaultShapeSize default symbol shape size in px
```
Comment (MV88) [inline web/client/plugins/Annotations.jsx:201]: also here is is missing defaultPointType 
```suggestion
  * @prop {string} defaultShapeSize default symbol shape size in px
  * @prop {string} defaultPointType default point type, can be "symbol" or "marker", default is "marker"

```

File: web/client/components/style/vector/Manager.jsx (hunk 1 above)
Comment (vlt1) [reply web/client/components/style/vector/Manager.jsx:217]: @MV88 This logic has been moved to epic to trigger when annotations are opened to have a central place to set styles with defaults from plugin. Also I think async logic should be in epics whenever possible.

**Comment by vlt1** [comment]
@MV88 Actually defaultPointType is from config, it is not a prop. That was my fault to include it in AnnotationsEditor as I didn't realize that the config is passed to it, so I changed code to access defaultPointType in AnnotationsEditor instead of Annotations component to pass it to Manager.

**Comment by MV88** [comment]
> @MV88 Actually defaultPointType is from config, it is not a prop. That was my fault to include it in AnnotationsEditor as I didn't realize that the config is passed to it, so I changed the code to access defaultPointType in AnnotationsEditor instead of Annotations component to pass it to Manager.

ok, i see, but i think we need to document how to configure the defaultPointType in the config property in localConfig.
probably  change to locale-config.md with this new property and all the missing ones if there are any

**Comment by tdipisa** [comment]
> > @MV88 Actually defaultPointType is from config, it is not a prop. That was my fault to include it in AnnotationsEditor as I didn't realize that the config is passed to it, so I changed the code to access defaultPointType in AnnotationsEditor instead of Annotations component to pass it to Manager.
> 
> ok, i see, but i think we need to document how to configure the defaultPointType in the config property in localConfig.
> probably change to local-config.md with this new property and all the missing ones if there are any

@MV88 @vlt1 remember to update also the plugin documentation if needed. We always need to keep this up do date with new config properties and related information.

File: web/client/components/mapcontrols/annotations/AnnotationsEditor.jsx [hunk 6]
```diff
@@ -191,6 +195,11 @@ class AnnotationsEditor extends React.Component {
         lineDashOptions: PropTypes.array,
         symbolList: PropTypes.array,
         defaultShape: PropTypes.string,
+        defaultShapeSize: PropTypes.number,
+        defaultShapeFillColor: PropTypes.string,
+        defaultShapeStrokeColor: PropTypes.string,
+        defaultPointType: PropTypes.string,
```
Comment (MV88) [inline web/client/components/mapcontrols/annotations/AnnotationsEditor.jsx:201]: should this be removed it defaultPointType comes from this.getConfig()?

File: web/client/components/mapcontrols/annotations/AnnotationsEditor.jsx [hunk 7]
```diff
@@ -317,6 +326,8 @@ class AnnotationsEditor extends React.Component {
                             onStartDrawing: this.props.onStartDrawing,
                             disabled: !this.props.config.multiGeometry && this.props.editing && this.props.editing.features && this.props.editing.features.length,
                             drawing: this.props.drawing,
+                            defaultStyles: this.props.defaultStyles,
+                            defaultPointType: this.props.defaultPointType,
```
Comment (MV88) [inline web/client/components/mapcontrols/annotations/AnnotationsEditor.jsx:330]: should this be changed like this?
```suggestion
                            defaultPointType: this.getConfig().defaultPointType,
```

File: web/client/components/style/vector/Manager.jsx [hunk 8]
```diff
@@ -64,25 +69,15 @@ class Manager extends React.Component {
...
-            color: "#000000",
-            fillColor: "#000000",
-            opacity: 1,
-            size: 64,
-            fillOpacity: 1
-        },
+        defaultShapeSize: 64,
+        defaultShapeFillColor: '#000000',
+        defaultShapeStrokeColor: '#000000',
+        defaultPointType: 'marker',
```
Comment (MV88) [inline web/client/components/style/vector/Manager.jsx:75]: to me the defaultPointType prop seems not used, if it is the case should be removed

File: web/client/components/style/vector/Manager.jsx [hunk 9]
```diff
@@ -52,6 +52,11 @@ class Manager extends React.Component {
         width: PropTypes.number,
         symbolsPath: PropTypes.string,
         defaultShape: PropTypes.string,
+        defaultShapeSize: PropTypes.number,
+        defaultShapeFillColor: PropTypes.string,
+        defaultShapeStrokeColor: PropTypes.string,
+        defaultPointType: PropTypes.string,
```
Comment (MV88) [inline web/client/components/style/vector/Manager.jsx:58]: check if this needs to be removed

**Pull request merged**